staging/src/k8s.io/apiserver/pkg/audit/context_test.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/apiserver/pkg/apis/audit"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+func TestWithAuditAnnotations(t *testing.T) {
+	var annotations []annotation
+	expContext := context.WithValue(context.Background(), auditAnnotationsKey, &annotations)
+
+	// Add key auditAnnotations to context
+	gotContext := WithAuditAnnotations(context.Background())
+	assert.Equal(t, expContext, gotContext)
+
+	// Key auditAnnotations is already in context
+	gotContext = WithAuditAnnotations(gotContext)
+	assert.Equal(t, expContext, gotContext)
+}
+
+func TestAddAuditAnnotation(t *testing.T) {
+	testCases := []struct {
+		desc                string
+		expEventAnnotations map[string]string
+		expCtxAnnotations   []annotation
+		getFunc             func(t *testing.T) (*audit.Event, []annotation)
```

Comment:
`t` seems to be unused in the instances of `getFunc`. You could make this be `modifyContext func(ctx context.Context) context.Context`, pull out the annotation and event retrieval outside and check it at the same spot in `t.Run()`